JSR223/Groovy “unable to resolve class org.openhab.binding.mqtt.internal.action.MQTTActions” with 4.3.0.M2

With OH 4.2.2 this automation/jsr223/z.groovy file
```groovy
var org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger("org.openhab.core.model.script.rele1")
var mqtt_broker = actions.get("mqtt", "mqtt:broker:b1")

logger.info('A')
logger.info(mqtt_broker.toString())
```
does work and prints:

> 2024-10-08 15:06:41.346 [INFO ] [ort.loader.AbstractScriptFileWatcher] - (Re-)Loading script '/etc/openhab/automation/jsr223/z.groovy'
> 2024-10-08 15:06:41.411 [INFO ] [org.openhab.core.model.script.rele1 ] - A
> 2024-10-08 15:06:41.421 [INFO ] [org.openhab.core.model.script.rele1 ] - org.openhab.binding.mqtt.internal.action.MQTTActions@12788b8

With OH 4.3.0.M2 it logs

> 2024-10-08 15:11:43.819 [INFO ] [ort.loader.AbstractScriptFileWatcher] - (Re-)Loading script '/etc/openhab/automation/jsr223/z.groovy'
> 2024-10-08 15:11:43.901 [ERROR] [ipt.internal.ScriptEngineManagerImpl] - Error during evaluation of script '/etc/openhab/automation/jsr223/z.groovy': org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
> /etc/openhab/automation/jsr223/z.groovy: -1: unable to resolve class org.openhab.binding.mqtt.internal.action.MQTTActions
> @ line -1, column -1.
> /etc/openhab/automation/jsr223/z.groovy: -1: unable to resolve class org.openhab.binding.network.internal.action.NetworkActions
> @ line -1, column -1.
> 2 errors
>
> 2024-10-08 15:11:43.903 [WARN ] [ort.loader.AbstractScriptFileWatcher] - Script loading error, ignoring file '/etc/openhab/automation/jsr223/z.groovy'

I do not know, if this is problem in JSR223/Automation or with Groovy.

--- a/bundles/org.openhab.automation.groovyscripting/src/main/java/org/openhab/automation/groovyscripting/internal/GroovyScriptEngineFactory.java
+++ b/bundles/org.openhab.automation.groovyscripting/src/main/java/org/openhab/automation/groovyscripting/internal/GroovyScriptEngineFactory.java
@@ -51,7 +51,15 @@ public class GroovyScriptEngineFactory extends AbstractScriptEngineFactory {
         ImportCustomizer importCustomizer = new ImportCustomizer();
         for (Map.Entry<String, Object> entry : scopeValues.entrySet()) {
             if (entry.getValue() instanceof Class<?> clazz) {
-                importCustomizer.addImport(entry.getKey(), clazz.getCanonicalName());
+                String canonicalName = clazz.getCanonicalName();
+                try {
+                    // Only add imports for classes that are available to the classloader
+                    getClass().getClassLoader().loadClass(canonicalName);
+                    importCustomizer.addImport(entry.getKey(), canonicalName);
+                    logger.debug("Added import for {} as {}", entry.getKey(), canonicalName);
+                } catch (ClassNotFoundException e) {
+                    logger.debug("Unable to add import for {} as {}", entry.getKey(), canonicalName, e);
+                }
             } else {
                 scriptEngine.put(entry.getKey(), entry.getValue());
             }

